Book-A-Desk.Api.Tests/FilteredBookingsTest.fs

```diff
@@ -0,0 +1,90 @@
+module Book_A_Desk.Api.Tests.FilteredBookingsTest
+
+open FsToolkit.ErrorHandling
+open System
+open Xunit
+
+open Book_A_Desk.Api.Models
+open Book_A_Desk.Domain.Office.Domain
+open Book_A_Desk.Infrastructure.DynamoDbEventStore
+open Book_A_Desk.Domain
+open Book_A_Desk.Infrastructure
+
+open Book_A_Desk.Domain.Events
+open Book_A_Desk.Domain.Reservation
+open Book_A_Desk.Domain.Reservation.Queries
+
+open Book_A_Desk.Domain.Reservation.Events
+
+
+let officeId = Guid.NewGuid ()
+let date = DateTime(2100,02,01)
+
+let createDeskBookedEvent date officeId email :DomainEvent =
+    ({
+            Date = date
+            EmailAddress = "anEmail" |> EmailAddress
+            OfficeId = officeId |> OfficeId
+    }: Events.DeskBooked) |> DeskBooked |> ReservationEvent
+let bookings: DomainEvent seq = 
+    [
+        createDeskBookedEvent date officeId "lasnikr"
+        createDeskBookedEvent date officeId "user"
+        createDeskBookedEvent date officeId "dummy"
+    ]
+        
+
+let mockProvideEventStore =
+        {
+            GetEvents = fun _ -> bookings |> Ok |> async.Return
+            AppendEvents = fun _ -> failwith "should not be called"
+        } : DynamoDbEventStore.DynamoDbEventStore
+
+let getBookingEvents : Option<seq<DomainEvent>> =
+    async {
+        let! result = mockProvideEventStore.GetEvents()
+        match result with
+        | Ok events -> return Some events
+        | Error errorMessage -> return None
+    }
+    |> Async.RunSynchronously
+
+let getBookingsFrom (bookingEvents: Option<seq<DomainEvent>>) (date : DateTime option)  (officeId : OfficeId option) (email : EmailAddress option) =
+    match bookingEvents with 
+    | Some bookingEvents -> 
+        let result = ReservationsQueriesHandler.getFilteredBookings bookingEvents date officeId email
+        match result with
+        | Ok bookings ->
+            let bookings =
+                bookings
+                |> List.map (fun (booking:Booking) ->
+                    Booking.value booking.OfficeId booking.Date booking.EmailAddress)
+                |> List.toArray
+                |> fun l -> { Bookings.Items = l }
+            bookings
+        | Error e -> failwith"Error occured"
+    | None -> failwith"Empty booking events"
+
+[<Fact>]
+let ``GIVEN booking events and a date WHEN filtering bookings THEN all bookings for the date are returned`` () = async {
+
```

Comment:
You generally want to isolate what's called your "system under test". In this case, this should be calling the API. It should be clear inside your test what is being tested. One way to achieve this is to have clear steps: a setup step, a testing step and an assert step. We generally call these "Arrange, Act and Assert". Unless you have a very long "Arrange" step, it's generally good practice to write your setup code, followed by an empty line, then the test and then the Asserts. 

Something like:
``` fsharp
[<Fact>]
let test () =
    let testSetup1 = 123
    let testSetup2 = 456

    let result = callingTheApi testSetup1 testSetup2

    Assert.Equal(expectedResult, result)
```